collector/node_collector.go

```diff
@@ -110,43 +112,39 @@ func (h *DCOSHost) RunPoller() {
 func (h *DCOSHost) pollHost() metricsMeta {
 	metrics := metricsMeta{}
 	now := time.Now().UTC()
-	nodeColLog.Infof("Polling the DC/OS host at %s:%d. Started at %s", h.IPAddress, h.Port, now.String())
+	metrics.timestamp = now.Unix()
+
+	// Fetch node-level metrics for all DC/OS roles
+	nodeColLog.Debugf("Fetching node-level metrics from DC/OS host %s:%d", h.IPAddress, h.Port)
+	nm, err := h.getNodeMetrics()
+	if err != nil {
+		nodeColLog.Errorf("Failed to get node-level metrics. %s", err)
+	} else {
+		metrics.nodeMetrics = nm
+	}
 
-	if h.DCOSRole == "agent" {
+	if h.DCOSRole == dcos.RoleAgent {
 		// always fetch/emit agent state first: downstream will use it for tagging metrics
 		nodeColLog.Debugf("Fetching state from DC/OS host %s:%d", h.IPAddress, h.Port)
 		agentState, err := h.getAgentState()
+
 		if err != nil {
 			nodeColLog.Errorf("Failed to get agent state from %s:%d. Error: %s", h.IPAddress, h.Port, err)
-			return metrics
+		} else {
+			metrics.agentState = agentState
```

Comment:
if this ok if we don't update `metrics.agentState`? this looks odd, maybe we should return error or something?